contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -154,10 +157,15 @@ pub fn execute_propose(
         return Err(ContractError::InactiveDao {});
     }
 
-    // Check that the sender is a member of the governance contract.
-    let sender_power = get_voting_power(deps.as_ref(), sender.clone(), config.dao.clone(), None)?;
-    if sender_power.is_zero() {
-        return Err(ContractError::MustHaveVotingPower {});
+    // Check whether open_propsoal_submissions are disabled
+    if !config.open_proposal_submission {
```

Comment:
how do we guarantee that the deposit has already been paid if we are here?

Reply:
maybe i'm misunderstanding but it seems like the flow is frontend --> execute_propose on prepropose module --> execute_propose on propose module --> new proposal hook on the prepropose module

but what if someone bypasses the frontend and executes execute_propose on the proposal_module? also why are the steps in execute_new_proposal_hook in the prepropose module not done during the deposit collection step?

Reply:
nvm, ignore this that is the permitted check, also this is the mc contract which is why I was confused